### PR TITLE
doc(runtime-c-api) Document and reorganize `exports.rs`.

### PR DESCRIPTION
**Pull request by Hywan**
This patch moves all structs at the beginning of the file, and
documents them.

### CODE REVIEW

**Comment by Hywan** [comment]
bors try

File: lib/runtime-c-api/src/export.rs [hunk 1]
```diff
@@ -241,7 +275,7 @@ pub unsafe extern "C" fn wasmer_export_func_params(
     }
 }
 
-/// Sets the returns buffer to the parameter types of the given wasmer_export_func_t
+/// Sets the returns buffer to the parameter types of the given wasmer_xport_func_t
```
Comment (syrusakbary) [inline lib/runtime-c-api/src/export.rs]: Perhaps this was not intended?

```suggestion
/// Sets the returns buffer to the parameter types of the given wasmer_export_func_t
```

File: lib/runtime-c-api/src/export.rs [hunk 2]
```diff
@@ -16,26 +16,63 @@ use std::{ptr, slice};
...
+/// exposed to C.
+pub(crate) struct NamedExports(pub Vec<NamedExport>);
+
+/// Opaque pointer to `NamedExports`.
 #[repr(C)]
 #[derive(Clone)]
-pub struct wasmer_export_func_t;
+pub struct wasmer_exports_t;
 
+/// Intermediate representation of an export dsscriptor that is
```
Comment (bjfish) [inline lib/runtime-c-api/src/export.rs]: typo in `dsscriptor`

Comment (syrusakbary) [reply lib/runtime-c-api/src/export.rs]:
```suggestion
/// Intermediate representation of an export descriptor that is
```

**Pull request merged**